### ISSUE TITLE
isset() check flagged as undefined variable

### ISSUE THREAD

**Issue by paulschreiber** [sirbrillig/phpcs-variable-analysis#202]
I would not expect and error from this code:

```php
21 if ( ! isset( $content_width ) ) {
22 	$content_width = 944;
23 }
```

```
FILE: functions.php
-------------------------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
-------------------------------------------------------------------------------------------------------------------------
 21 | WARNING | Variable $content_width is undefined.
    |         | (VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable)
-------------------------------------------------------------------------------------------------------------------------
```

**Comment by sirbrillig** [comment]
Ah, good point! `isset` and `empty` should probably both be allowed to specify a variable that doesn't exist.

**Comment by sirbrillig** [comment]
A tangent to this is that reading an undefined variable after a successful `isset()` or `! empty()` check should not be reported as undefined. That is,

```php
if (isset($undefinedVar)) { // no warning
    doSomething($undefinedVar); // also no warning
}
```

This is tricky and I'm not sure if I can do it though because while it's easy to see that in the above example `$undefinedVar` must be defined, what about this?

```php
if ($override || isset($undefinedVar)) { // no warning
    doSomething($undefinedVar); // probably should still be a warning
}
```

It could get quite complex to tell if the variable is defined or not after the `isset`. And the logic necessary to do so is getting pretty close to static analysis territory, which is better handled by something like psalm or phpstan. So unfortunately I don't think we'll be able to add that here, but I thought it was worth mentioning (maybe you or someone else can come up with a more foolproof way to figure it out).

The example you gave is fine, though, since line 22 is a write rather than a read, and those are quite safe to use with an undefined variable.
